libcamera-rs/src/camera_manager.rs

```diff
@@ -1,25 +1,39 @@
-use std::{ffi::CStr, marker::PhantomData, ptr::NonNull};
+use std::{ffi::CStr, io, marker::PhantomData, ptr::NonNull};
 
 use libcamera_sys::*;
+use thiserror::Error;
 
 use crate::camera::Camera;
 
+#[derive(Debug, Error)]
+pub enum CameraManagerError {
+    #[error("No cameras were found by the camera manager")]
+    NoCamerasFound,
+    #[error(transparent)]
+    Unexpected(#[from] io::Error),
+}
+
+impl CameraManagerError {
+    pub fn from_raw_os_error(errno: i32) -> Result<(), Self> {
+        match errno {
+            e if e >= 0 => Ok(()),
+            e if e == -libc::ENODEV => Err(Self::NoCamerasFound),
```

Comment:
I don't think camera manager can (or should) return this error. `CameraManager::new()` always succeeds (unless there is another unexpected error) and if there are no cameras, `CameraManager::cameras().len()` is zero.